dnsutil/util_test.go

```diff
@@ -0,0 +1,130 @@
+package dnsutil
+
+import "testing"
+
+// func AddOrigin
+
+func TestAddOrigin(t *testing.T) {
+	var tests = []struct{ e1, e2, expected string }{
+		{"@", "example.com", "example.com"},
+		{"foo", "example.com", "foo.example.com"},
+		{"foo.", "example.com", "foo."},
+		{"@", "example.com.", "example.com."},
+		{"foo", "example.com.", "foo.example.com."},
+		{"foo.", "example.com.", "foo."},
+		// Oddball tests:
+		// In general origin should not be "" or "." but at least
+		// these tests verify we don't crash and will keep results
+		// from changing unexpectedly.
+		{"*.", "", "*."},
+		{"@", "", "@"},
+		{"foobar", "", "foobar"},
+		{"foobar.", "", "foobar."},
+		{"*.", ".", "*."},
+		{"@", ".", "."},
+		{"foobar", ".", "foobar."},
+		{"foobar.", ".", "foobar."},
+	}
+	for _, test := range tests {
+		actual := AddOrigin(test.e1, test.e2)
+		if test.expected != actual {
+			t.Errorf("AddOrigin(%#v, %#v) expected %#v, go %#v\n", test.e1, test.e2, test.expected, actual)
+		}
+	}
+}
+
+// func TrimDomainName
+
+func TestTrimDomainName(t *testing.T) {
+	var tests = []struct{ experiment, expected string }{
+		{"foo.example.com", "foo"},
+		{"foo.example.com.", "foo"},
+		{"*.example.com", "*"},
+		{"serverfault.com.", "serverfault.com."},
+		{"com.", "com."},
+		{"example.com", "@"},
+		{"example.com.", "@"},
+		{"foo.", "foo."},
+		{".foo.example.com", ".foo"},
+		{".foo.example.com.", ".foo"},
+		{".foo.ronco.com", ".foo.ronco.com"},
+		{".foo.ronco.com.", ".foo.ronco.com."},
+		{"serverfault.com", "serverfault.com"},
+	}
+	for _, dom := range []string{"example.com", "example.com."} {
+		for i, test := range tests {
+			actual := TrimDomainName(test.experiment, dom)
+			if test.expected != actual {
+				t.Errorf("%d TrimDomainName(%#v, %#v): expected (%v) got (%v)\n", i, test.experiment, dom, test.expected, actual)
+			}
+		}
+	}
+}
+
+func TestTrimDomainNameShortSuffix(t *testing.T) {
```

Comment:
Could you collopse this into 2 two test function for both AddOrigin and TrimDomainName